blocks/algorithms/__init__.py

```diff
@@ -169,6 +170,10 @@ def __init__(self, step_rule=None, gradients=None, **kwargs):
                 zip(self.params, tensor.grad(self.cost, self.params))))
         self.step_rule = step_rule if step_rule else SteepestDescent()
 
+        self.total_gradient_norm = named_copy(
```

Comment:
Yep, I also had it mind. I see no problem in adding it a bit later, and seems we can release 0.01 without it.